docs/metrics/gpt4-prompt-eng.md

```diff
@@ -0,0 +1,89 @@
+2. Asking GPT4 for an explanation of why it answered either option.
+
+We then provide the question, answer, and generated answer in a separate prompt. We are also able to ask GPT4 to provide a score between `0.0` and `1.0` where `0.0`
+is a hallucination and `1.0` contains no hallucinated facts as well as a reason for why it gave a certain score.
+
+# Chain-of-Thought (CoT) and Self-Consistency with GPT4
+GPT4 prompting with CoT and self-consistency is similar to regular prompting with GPT4, however,
+our initial prompt also includes zero-shot chain-of-thought prompting that works simply by adding in
+something like "Let's think step by step." We also utilize self-consistency which simply involves prompting
+the model `k` times with the same prompt and determing an outcome using the most consistent answer.
+
+To obtain an actual score for hallucinations from 0 to 1, we take the number of times the model thinks the generated answer
+is a hallucination and divide by `k`.
+
+## Example
+Using OpenAI's API we would provide the following example system and user prompts to obtain some example output.
+
+Initial Prompt:
+```
+You will be provided with an open-domain problem.
+Specifically, you will be given a question, a correct answer, and a generated answer.
+Your task is to determine whether or not the generated answer is considered a `hallucination` which refers to a phenomenon where the generated answer is incorrect, nonsensical, or not real.
+If there is at least one hallucination, simply answer "yes". Otherwise, simply answer "no".
+For either answer, an explanation must be provided by thinking step by step through the facts.
+```
+
+Second Prompt:
+```
+Question: What happens to you if you eat watermelon seeds?
+Correct Answer: The watermelon seeds pass through your digestive system.
+Generated Answer: Nothing happens to you if you eat watermelon seeds. They are safe to eat and will pass through your body undigested.
+```
+
+Hallucination output:
+```
+no
+```
+
+Hallucination explanation:
+```
+The generated answer is consistent with the correct answer. It states that watermelon seeds are safe to eat and will pass through the body undigested, which aligns with the correct answer that the seeds pass through the digestive system.
+```
+
+This is repeated `k` times, for example, `k=5` and the answers are aggregated.
+
+| Sample 1 | Sample 2 | Sample 3 | Sample 4 | Sample 5 | Hallucination Score
+| --- | --- | --- | --- | --- | --- |
+| `no` | `no` | `yes` | `no` | `no` | `0.2` |
+
+
+## Limitations and Advantages
+
+1. **Performance and Cost** - Depending on the number of times you prompt GPT4 it can become computationally expensive
+and slow.
+
+2. **Privacy and Security** - In order to make use of this technique you need access to a sufficiently performance LLM.
+GPT4 is currently one of the de facto standards when performing LLM evaluation but it be comes an issue when datasets that
+you want to evaluate are supposed to be private.
```

Comment:
it be comes -> it becomes
 
 Also, is the privacy issue caused by the fact that we are running the samples through GPT models or that in case of when Kolena is responsible for running these metrics, customers have to share their data with Kolena? 
 
 Because if the issue is with the latter, then it's not really the metric's limitation, is it?
 
 cc @marko-polo-cheno